src/napari_graph/_base_graph.py

```diff
@@ -546,6 +546,18 @@ def _validate_edges(self, edges: ArrayLike) -> np.ndarray:
 
         return edges
 
+    @abstractmethod
+    def _add_edge(self, src_node: int, tgt_node: int) -> None:
+        raise NotImplementedError
+
+    def add_edge(self, src_node: int, tgt_node: int) -> None:
```

Comment:
To reduce duplicated code I would remove the `_add_edge` method and `add_edge` would only wrap a `add_edges` call, for example:

```python3
def add_edge(see, src_node: int, tgt_node: int) -> None:
    self.add_edges([[src_node, tgt_node]])
```

Since it's a single node that's being inserted I don't think performance is critical, otherwise, it would be using the `add_edges` method.

The same is valid for `remove_edge`.

Reply:
Looking at this again, I agree. Actually, I'm not so sure anymore whether we should actually provide functions for adding/removing single nodes/edges :see_no_evil: Thinking about it, this could potentially incentivize inefficient programming (e.g. adding nodes in a loop), and just calling the corresponding multi-item functions like in your example is probably simple enough. What do  you think? Should we keep the single-node/edge functions?

Reply:
I agree with you. We can keep just the renaming.

Reply:
How about the `add_node`/ `remove_node` methods? Should they also be changed to `add_nodes`/`remove_nodes` for consistency?

Reply:
Yes, I think we could change it, I would still keep it in python (not numba), though.

Reply:
In the interest of time, leaving the `add_node`/`remove_node` methods as they are for now